design-documents/testing/functional/test-package.md

```diff
@@ -0,0 +1,96 @@
+It means that all MFTF tests currently also distributed as a part of `magento2-module`.
+When SIs or Extension developers installs Magento via Composer they are able to run and use MFTF tests for their own needs.
+
+### Problem
+This approach leads to situation where production environment always contains code (in our case xml configurations) which will never be used.
+
+Also here is reasons described by Piotr Kaminski why it is not good to have tests on prod env.
+>There are couple reasons to separate tests from core distribution:
+>
+>Security:
+> - reduce possible attack surface (less code)
+> - the test code is not written with security in mind
+> - even if we protect the access via .htaccess, some sites run Nginx or skip .htaccess making this code accessible
+> - we had security issues in the past that resulted in remote code exec using code related to tests
+>
+>Future-proof:
+> - ability to release separately, on separate cadence
+> - matches the general plan for all other non-critical modules - we want to modularize and release separately things like payment extensions, shipping and more.
+
+
+### Proposed solution
+During Magento package new releases automated tool will exclude tests from main `magento2-module` package and create other package with type `magento2-test-module`.
+
+## Packaging flows
+### General Flow
+1. Exclude `Test` folder from module and keep it for further packaging flow as it shown on image below.
+
+![Packaging Process](img/packaging.png)
+
+2. Create `magento2-module` package as usual, but without tests.
+
+3. Out of excluded tests create a separate package.
+   - For this package we will automatically generate `composer.json` based on existing module `composer.json` file as shown on image below.
+    
+![Composer file](img/test_composer_json.png)
+
+### Version setter flow #1
+1. Versions will be set the same as for `magento2-module`. It means if we release `magento/module-store` version `101.0.3` we will specify same version for `magento/test-module-store`.
+2. Metapackage `magento/product-tests-community-edition` and `magento/project-tests-community-edition` created and have the same versions as Magento metapackage. 
+
+#### Benefits:
+1. Very easy to understand what version of tests you need to run against existing Magento instance
+2. Publication build will have small changes to enable this approach.
+
+#### Disadvantages: 
+1. Can not make separate releases for module or test. Always 2 packages will be released.
+
+### Version setter flow #2
+1. Versions will be set independently from Magento module based on SVC tool results.
+2. Metapackage `magento/product-tests-community-edition` and `magento/project-tests-community-edition` created and have the same versions as Magento metapackage.
+
+#### Benefits:
+1. Possibility to release test package only
+
+#### Disadvantages: 
+1. Bigger changes to Publication build
```

Comment:
MFTF versioning proposal becomes obsolete because tests version bump must always be aligned with Magento. Also, it is possible that tests will have breaking changes while the module itself will have PATCH changes.
Does not seem like this is even an option.